### DISCUSSION TITLE
self-hosting with apache2 on raspberry pi

### DISCUSSION THREAD

**Discussion by amychurchwell** [firefly-iii/firefly-iii#4788] [Firefly III] [answered]
✅ I have followed the directions in the docs on [self-hosted installation ](https://docs.firefly-iii.org/firefly-iii/installation/self_hosted/) and in the [FAQ about getting a 404](https://docs.firefly-iii.org/firefly-iii/faq/self_hosted/#i-get-a-404)

Not sure how to move forward. /firefly-iii/ always 404's whenever I navigate to it. I am wondering if I have somehow configured AllowOverride incorrectly...

`/etc/apache2/apache2.conf`:
```
<Directory />
        Options FollowSymLinks
        AllowOverride None
        Require all denied
</Directory>

<Directory /usr/share>
        AllowOverride None
        Require all granted
</Directory>

<Directory /var/www/>
        Options Indexes FollowSymLinks
        AllowOverride All
        Require all granted
</Directory>
```

`/etc/apache2/sites-available/000-default.conf`:
```
<VirtualHost *:80>
        ServerAdmin webmaster@localhost
        DocumentRoot /var/www/html

        <Directory /var/www/>
            Options Indexes FollowSymLinks MultiViews
            AllowOverride All
            Require all granted
        </Directory>
</VirtualHost>
```

**Comment by JC5** [comment, marked as answer]
Set document root to /var/www/html/firefly-iii/public

**Reply by amychurchwell** [reply]
Thanks for the reply! This worked! \o/
However, doesn't setting firefly up as root mean I can't host other apps or websites? The docs say to create the project in `/var/www` and this required me to move the firefly-iii folder to `/html` which means it's not served at `http://localhost/firefly-iii/`

**Reply by JC5** [reply]
You'll have to work with virtual hosts and subdomains yes, if you want to host multiple sites on the same host.

You are free to keep Firefly III in /var/www, but you would have to change the paths I just pointed out. It really doesn't matter.

**Reply by amychurchwell** [reply]
Okay, thanks. I can make a PR to update the docs to reflect this so others don't run into the same issue.

**Reply by JC5** [reply]
About what, exactly?

**Reply by amychurchwell** [reply]
About how if you are [using apache and get a 404](https://docs.firefly-iii.org/firefly-iii/faq/self_hosted/#i-get-a-404) you need to change the document root.

**Reply by JC5** [reply]
Ah yes, feel free to submit a PR, thanks!
